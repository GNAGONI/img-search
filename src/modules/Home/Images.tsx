import { useSelector } from 'react-redux';
import { getHomeImages } from './state/selectors';
import { styled } from '@mui/material/styles';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import Image from '../../components/Image';

const ImagesWrapper = styled('div')({
  padding: '16px',
})

const Images = () => {
  const images = useSelector(getHomeImages);
  return (
    <ImagesWrapper>
      <Box sx={{ flexGrow: 1 }}>
        <Grid
          container
          direction="row"
          justifyContent="center"
          alignItems="center"
          spacing={2}
        >
          {images.map((image) => (
            <Grid item xs="auto" sm="auto" md="auto" key={image.id}>
              <Image url={image?.urls?.small} alt={image?.alt_description} />
            </Grid>
          ))}
        </Grid>
      </Box>
    </ImagesWrapper>
  );
};

export default Images;
