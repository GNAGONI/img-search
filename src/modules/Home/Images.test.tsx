import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import { render, screen } from '@testing-library/react';
import Images from './index';

jest.mock("react-redux", () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

describe('Images', () => {
  const useSelectorMock = reactRedux.useSelector;
  const mockStore = {
    home: {
      images: [
        { id: 1, url: 'www.1.com', alt: "alt img1" },
        { id: 2, url: 'www.2.com', alt: "alt img2" },
        { id: 3, url: 'www.2.com', alt: "alt img3" }
      ]
    }
  };
  beforeEach(() => {
    (useSelectorMock as jest.Mock).mockImplementation(selector => selector(mockStore));
  });
  afterEach(() => {
    (useSelectorMock as jest.Mock).mockClear();
  });

  it('should have a valid snapshot', () => {
    const tree = renderer
      .create(
        <Images />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should behave correctly', () => {
    render(
      <Images />
    );

    const image1 = screen.getByAltText('alt img1',);
    expect(image1).toBeInTheDocument();
    expect(image1).toBeVisible();

    const image2 = screen.getByAltText('alt img2',);
    expect(image2).toBeInTheDocument();
    expect(image2).toBeVisible();

    const image3 = screen.getByAltText('alt img3',);
    expect(image3).toBeInTheDocument();
    expect(image3).toBeVisible();
  });
});

