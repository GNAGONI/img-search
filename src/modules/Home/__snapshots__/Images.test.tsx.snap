// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images should have a valid snapshot 1`] = `
<div>
  <form
    className="css-epn0vg"
    onSubmit={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-t7h6ca-MuiFormControl-root-MuiTextField-root"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink={false}
        htmlFor=":r0:"
        id=":r0:-label"
      >
        Search
        <span
          aria-hidden={true}
          className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
        >
           
          *
        </span>
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          id=":r0:"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={true}
          type="text"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-yjsfm1"
          >
            <span>
              Search
               
              *
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-q1b40h-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <p>
        Search
      </p>
    </button>
  </form>
  <div
    className="css-18o18yt"
  >
    <div
      className="MuiBox-root css-i9gxme"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-6s1xft-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto MuiGrid-grid-sm-auto MuiGrid-grid-md-auto css-1wrgmsj-MuiGrid-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
          >
            <button
              className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <img
                alt="alt img1"
                className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="300"
                src="www.1.com"
              />
              <span
                className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto MuiGrid-grid-sm-auto MuiGrid-grid-md-auto css-1wrgmsj-MuiGrid-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
          >
            <button
              className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <img
                alt="alt img2"
                className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="300"
                src="www.2.com"
              />
              <span
                className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto MuiGrid-grid-sm-auto MuiGrid-grid-md-auto css-1wrgmsj-MuiGrid-root"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ri6ub7-MuiPaper-root-MuiCard-root"
          >
            <button
              className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <img
                alt="alt img3"
                className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
                height="300"
                src="www.2.com"
              />
              <span
                className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-19midj6"
    >
      <div
        className="css-1p5q5e5-MuiStack-root"
      >
        <nav
          aria-label="pagination navigation"
          className="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
        >
          <ul
            className="MuiPagination-ul css-wjh20t-MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateBeforeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-current="true"
                aria-label="page 1"
                className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-selected MuiPaginationItem-page css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                1
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                className="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-textPrimary Mui-disabled MuiPaginationItem-previousNext css-1to7aaw-MuiButtonBase-root-MuiPaginationItem-root"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateNextIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
